src/main/java/de/zalando/ep/zalenium/container/kubernetes/KubernetesContainerClient.java

```diff
@@ -610,16 +639,24 @@ public static DoneablePod createDoneablePodDefaultImpl(PodConfiguration config)
                         .endResources()
                         // Add a readiness health check so that we can know when the selenium pod is ready to accept requests
                         // so then we can initiate a registration.
+                        .withNewLivenessProbe()
+                            .withInitialDelaySeconds(30)
+                            .withTimeoutSeconds(5)
+                            .withNewHttpGet()
+                                .withPath("/wd/hub/status")
+                                .withPort(new IntOrString(40000))
+                            .endHttpGet()
+                        .endLivenessProbe()
                         .withNewReadinessProbe()
-                            .withNewExec()
-                                .addToCommand(new String[] {"/bin/sh", "-c", "http_proxy=\"\" curl -s http://`hostname -i`:"
-                                        + config.getNodePort() + "/wd/hub/status | jq .value.ready | grep true"})
-                            .endExec()
                             .withInitialDelaySeconds(5)
-                            .withFailureThreshold(60)
-                            .withPeriodSeconds(1)
-                            .withTimeoutSeconds(5)
+                            .withFailureThreshold(2)
+                            .withPeriodSeconds(2)
+                            .withTimeoutSeconds(10)
                             .withSuccessThreshold(1)
+                            .withNewHttpGet()
```

Comment:
We would still need to know if the JSON payload from the status endpoint returns ready or not, right?

Reply:
Yes, but we have less problem using this way to check the status although there are cases where the grid can not be ready. . With high concurrency the readiness didn't work properly.

Reply:
But the only way to know if the node is ready is to query the endpoint, if the enpoint is not ready then the node cannot receive any tests. Am I missing something?